tools/spectral/ipa/rulesets/functions/listMethodResponseIsGetMethodResponse.js

```
@@ -0,0 +1,115 @@
+    (isResourceCollectionIdentifier(resourcePath) && isSingletonResource(getResourcePathItems(resourcePath, oas.paths)))
+  ) {
+    return;
+  }
+
+  // Ignore if the List method does not have a response schema
+  const listMethodResponse = resolveObject(oas, path);
+
+  if (!listMethodResponse || !listMethodResponse.schema) {
+    return;
+  }
+
+  if (hasException(listMethodResponse, RULE_NAME)) {
+    collectException(listMethodResponse, RULE_NAME, path);
+    return;
+  }
+
+  // Get list response schema from ref or inline schema
+  let resolvedListSchema;
+  const listSchemaRef = getSchemaRef(listMethodResponse.schema);
+  if (!listSchemaRef) {
+    resolvedListSchema = listMethodResponse.schema;
+  } else {
+    const listSchemaName = getSchemaNameFromRef(listSchemaRef);
+    resolvedListSchema = resolveObject(oas, ['components', 'schemas', listSchemaName]);
+  }
+
+  // Ignore if the List method response is not found or not paginated
+  if (!resolvedListSchema || !schemaIsPaginated(resolvedListSchema)) {
+    return;
+  }
+
+  // Ignore if there is no matching Get method
+  const getMethodResponseContentPerMediaType = getResponseOfGetMethodByMediaType(mediaType, resourcePath, oas);
+  if (!getMethodResponseContentPerMediaType) {
+    return;
+  }
+
+  const errors = checkViolationsAndReturnErrors(
+    path,
+    resolvedListSchema.properties.results.items,
+    getMethodResponseContentPerMediaType
+  );
+
+  if (errors.length !== 0) {
+    return collectAndReturnViolation(path, RULE_NAME, errors);
+  }
+
+  collectAdoption(path, RULE_NAME);
+};
+
+function checkViolationsAndReturnErrors(path, listMethodResultItems, getMethodResponseContent) {
+  try {
+    // Error if the Get method does not have a schema
+    if (!getMethodResponseContent.schema) {
+      return [
+        {
+          path,
+          message: `Could not validate that the List method returns the same resource object as the Get method. The Get method does not have a schema.`,
```

Q: Should we return the same error for the createMethodRequest comparison?

---

Sure, that makes sense 👍